tests/cases/conformance/jsdoc/callbackOnConstructor.ts

```diff
@@ -0,0 +1,17 @@
+// @filename: callbackOnConstructor.js
+// @checkJs: true
+// @outdir: dist
+// @declaration: true
+export class Preferences {
+  assignability = "no"
+  /**
+   * @callback ValueGetter_2
+   * @param {string} name
+   * @returns {boolean|number|string|undefined}
+   */
+  constructor() {}
+}
+
+
+/** @type {ValueGetter_2} */
```

Comment:
Should this include all of the examples from the issue, just so we have them, even if the third one is still broken?

Reply:
No, I opened #54037 to track that, and the second bug also occurs on EmptyStatement (or whatever it's parsed as inside a class body).